src/ring_theory/multiplicity.lean

```diff
@@ -260,6 +260,23 @@ begin
   rwa zero_dvd_iff,
 end
 
+lemma multiplicity_eq_multiplicity_mk [decidable_rel ((∣) : associates α → associates α → Prop)]
+  {a b : α} : multiplicity a b = multiplicity (associates.mk a) (associates.mk b) :=
+begin
```

Comment:
I feel like this should be easier since multiplicity is defined only in terms of divisibility and powers, and `associates.mk` preserves both. Unfortunately, the only way I can figure out to actually prove that is to define the theory of monoid homomorphisms which are divisibility-congruent (that is, a thousand lines of code to replace these 10)...

Reply:
I originally wanted to prove the lemma doing something along that train of thought, but I quickly realised it would be rather involved :-)